### PR TITLE
feat: add telemetry settings to configure health check ports

### PR DESCRIPTION
**Pull request by hessjcg**
Allow the user to configure the admin and health check ports for a proxy container. This adds TelemetrySpec
which will hold the ports as well as other telemetry-related configuration.

Related to #45

### CODE REVIEW

File: config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml [hunk 1]
```diff
@@ -902,6 +902,18 @@ spec:
                     sqlAdminAPIEndpoint:
                       description: SQLAdminAPIEndpoint is a debugging parameter that when specified will change the Google Cloud api endpoint used by the proxy.
                       type: string
+                    telemetry:
```
Comment (hessjcg) [inline config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml:905]: Generated from authproxyworkload_types.go

File: installer/cloud-sql-proxy-operator.yaml [hunk 2]
```diff
@@ -920,6 +920,18 @@ spec:
                     sqlAdminAPIEndpoint:
                       description: SQLAdminAPIEndpoint is a debugging parameter that when specified will change the Google Cloud api endpoint used by the proxy.
                       type: string
+                    telemetry:
```
Comment (hessjcg) [inline installer/cloud-sql-proxy-operator.yaml:923]: Generated from authproxyworkload_types.go

File: internal/workload/podspec_updates_test.go [hunk 3]
```diff
@@ -622,6 +623,34 @@ func TestProxyCLIArgs(t *testing.T) {
...
+					ConnectionString: "hello:world:one",
+				}},
+			},
+			wantProxyArgContains: []string{
+				fmt.Sprintf("hello:world:one?port=%d", workload.DefaultFirstPort),
+			},
+			wantWorkloadEnv: map[string]string{
+				"CSQL_PROXY_SQLADMIN_API_ENDPOINT": "https://example.com",
+				"CSQL_PROXY_HTTP_PORT":             "9092",
+				"CSQL_PROXY_ADMIN_PORT":            "9091",
```
Comment (enocom) [inline internal/workload/podspec_updates_test.go:648]: Setting the admin port alone won't enable the admin server. 

See https://github.com/GoogleCloudPlatform/cloud-sql-proxy#localhost-admin-server.

Should we make admin-related changes a separate PR?

Comment (hessjcg) [reply internal/workload/podspec_updates_test.go:648]: Yes. I'll remove the ADMIN_PORT envvar from this PR and move it into another one.

File: internal/workload/podspec_updates.go [hunk 4]
```diff
@@ -44,6 +44,10 @@ const (
 	// DefaultHealthCheckPort is the used by the proxy to expose prometheus
 	// and kubernetes health checks.
 	DefaultHealthCheckPort int32 = 9801
+
+	// DefaultAdminPort is the used by the proxy to expose prometheus
+	// and kubernetes health checks.
+	DefaultAdminPort int32 = 9802
```
Comment (enocom) [inline internal/workload/podspec_updates.go:50]: I think this can go since it's a duplicate of the field above?

Comment (hessjcg) [reply internal/workload/podspec_updates.go]: Fixed.

**Review by enocom — APPROVED** [review]

**Pull request merged**